sys/dev/pci/pci_user.c

```diff
@@ -1403,6 +1427,18 @@ pci_ioctl(struct cdev *dev, u_long cmd, caddr_t data, int flag, struct thread *t
 		break;
 
 	case PCIOCBARMMAP:
+#ifdef COMPAT_FREEBSD64
+	case PCIOCBARMMAP64:
+		if (cmd == PCIOCBARMMAP64) {
+			pbm64 = (struct pci_bar_mmap64 *)data;
+			pbm = &pbms;
+			pbm->pbm_map_base = __USER_CAP(pbm64->pbm_map_base,
+			    pbm64->pbm_map_length);
+			memcpy(__unbounded_addressof(pbm->pbm_map_length),
+			    __unbounded_addressof(pbm64->pbm_map_length),
+			    sizeof(struct pci_bar_mmap64) - sizeof(uint64_t));
```

Comment:
```suggestion
			    sizeof(struct pci_bar_mmap64) - offsetof(struct pci_bar_mmap64, pbm_map_length);
```

Reply:
That being said, the usual style would be to copy individual fields with
```C
CP(pbm64, pbm, pbm_map_length);
CP(pbm64, pbm, pbm_sel);
...
```

Reply:
Moved to `CP()` style copy.